platform/host_api.go

```diff
@@ -819,37 +801,51 @@ func (bh *BraveHost) InitUnit(backend Backend, unitParams *shared.Bravefile) err
 		return err
 	}
 
+	abortFlag := false
+	ctx := context.Background()
+	ctx, cancel := context.WithCancel(ctx)
+
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		for range c {
+			fmt.Println("Interrupting build and cleaning artefacts")
+			abortFlag = true
+			cancel()
+		}
+	}()
+
 	fingerprint, err = ImportImage(image, unitParams.PlatformService.Name, bh.Remote)
-	if err != nil {
+	if err != nil || abortFlag {
 		return errors.New("failed to import image: " + err.Error())
```

Comment:
When interrupting during Image Import, this line causes panic:

"""bash
⠧ Importing mlbase-1.0.tar.gz ^CInterrupting build and cleaning artefacts
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x4528ded]

goroutine 1 [running]:
github.com/bravetools/bravetools/platform.(*BraveHost).InitUnit(0x4bf7d20, {0x486e930, 0xc0002780f0}, 0xc0001c5540)
	/Users/ignat/go/src/github.com/beringresearch/bravetools/platform/host_api.go:820 +0x2ad
github.com/bravetools/bravetools/commands.deploy(0x4bed440?, {0x4c28890, 0x0, 0x0?})
	/Users/ignat/go/src/github.com/beringresearch/bravetools/commands/deploy.go:98 +0x35f
github.com/spf13/cobra.(*Command).execute(0x4bed440, {0x4c28890, 0x0, 0x0})
	/Users/ignat/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x4becf40)
	/Users/ignat/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x39c
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/ignat/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
main.main()
	/Users/ignat/go/src/github.com/beringresearch/bravetools/main.go:15 +0x25
"""

I think this is because `err` is actually nil and the `if` statement is trying to return `err.Error()`

Reply:
That makes sense. Will have to take a different approach to err checking - likely delegate it to a helper function and reuse throughout.

Reply:
This issue should now be dealt with. The shared utility function `CollectErrors` returns an error if there is one, so there will no longer be a case where a nil error is unexpectedly used.